src/components/Dialogs/Conditions/ConditionsManager.js

```diff
@@ -73,15 +75,15 @@ const ConditionsManager = ({
     onClose,
     setConditionsByDimension,
 }) => {
-    const dimensionType = dimension.valueType
+    const valueType = dimension.valueType
     const isOptionSetCondition =
-        dimensionType === DIMENSION_TYPE_TEXT && dimension.optionSet
+        valueType === DIMENSION_TYPE_TEXT && dimension.optionSet
 
     const [conditionsList, setConditionsList] = useState(
         (conditions.condition?.length &&
             parseConditionsStringToArray(conditions.condition)) ||
             (!conditions.condition?.length &&
-                (SINGLETON_TYPES.includes(dimensionType) ||
+                (SINGLETON_TYPES.includes(valueType) ||
```

Comment:
Just wondering why it's called "singleton"

Reply:
I'm not able to read through this multi-line check and understand what it does.

Reply:
`Singleton` as in _can only be instantiated once_. Some dimensions can have multiple conditions (GT:10;LT:15 etc). They go on separate rows / instances of the corresponding component (e.g. `NumericCondition.js`). However some conditions are single-use, e.g. booleans that will display some checkboxes. You can never display the set of checkboxes multiple times, hence they're "singleton".

Reply:
> I'm not able to read through this multi-line check and understand what it does.

No, I agree, it's not super readable. Then again, the logic is quite complex. I could for example extract parts of it separate consts for this purpose only, but I'm not sure if that would make it easier to understand. The key problem is that ConditionsManager handles a lot of different conditions, which themselves vary greatly. Option set, legend set, singleton types and the "regular" types, so it quickly becomes pretty hard to read regardless I think...